File: 14.longest-common-prefix.cpp

#include <vector>
#include <string>
#include <iostream>

using namespace std;
/*
 * @lc app=leetcode id=14 lang=cpp
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
class Solution
{
public:
    string longestCommonPrefix(const vector<string>& strs) {
        string pref;
        for (int i = 0; i < strs[0].size(); ++i) {
            for (auto &s : strs) {
                if (s.size() == i || s[i] != strs[0][i]) {
                    return pref;
                }
            } 
            pref += strs[0][i];
        }
        return pref;
    }
};
// @lc code=end

int main()
{
    cout << Solution{}.longestCommonPrefix({"asdf", "a"}) << endl;
    return 0;
}

File: 14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#

from typing import List

# @lc code=start

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        a = ""
        for i, ch in enumerate(strs[0]):
            for s in strs:
                if len(s) == i or s[i] != ch:
                    return a
            a += ch
        return a
        
# @lc code=end

print(Solution().longestCommonPrefix(["abc123", "ab", "abcf"]))

